SquAPI.lua

```diff
@@ -73,1100 +68,1238 @@ squapi.autoFunctionUpdates = true
+  end
 
-    table.insert(squapi.arms, self)
-    return self
+  table.insert(squapi.arms, _self)
+  return _self
 end
 
-
-
--- SMOOTH HEAD - Mimics a vanilla player head, but smoother and with some extra life. can also do smooth Torsos and Smooth Necks!
--- guide:(note if it has a * that means you can leave it blank to use reccomended settings)
--- element: 			 The head element that you wish to effect
--- *strength:            The target rotation is multiplied by this factor. For example setting to 1 will follow vanilla rotation, 0.5 is half of that, and 2 is double vanilla rotation. 
--- *tilt:                For context the smooth head applies a slight tilt to the head as it's rotated toward the side, this controls the strength of that tilt.  
--- *speed:               How fast the head will rotate toward the target rotation. For example 1 is base speed, 0.5 is half of that, and 2 is double speed. 
--- *keepOriginalHeadPos: When true(automatically true) the heads position will follow the vanilla head position. For example when crouching the head will shift down to follow. set to false to disable.
-
---Smooth Neck? Smooth Torso?
---This can do that too if you change what you input for these:
--- element:     Instead of a single element, input a table of head elements(imagine it like {element1, element2, etc.}). This will apply the head rotations to each of these.
--- *strength:   Instead of an single number, you can put in a table(imagine it like {strength1, strength2, etc.}). This will apply each strength to each respective element.(make sure it is the same length as your element table)
--- As a tip, you can imagine the strength as a percentage of the heads vanilla rotation. 
--- So if you have a head and a torso, you might do 0.5 for the head, and 0.5 for the torso to add up to 1(100% of the vanilla heads rotation), or maybe even 0.25 for torso, and 0.75 for head, it's up to you!
-
+---Contains all registered smooth heads
+---@type SquAPI.SmoothHead[]
 squapi.smoothHeads = {}
 squapi.smoothHead = {}
 squapi.smoothHead.__index = squapi.smoothHead
-function squapi.smoothHead:new(element, strength, tilt, speed, keepOriginalHeadPos)
-    local self = setmetatable({}, squapi.smoothHead)
-	
-    -- INIT -------------------------------------------------------------------------
-    if type(element) == "ModelPart" then
-        assert(element, "§4Your model path for smoothHead is incorrect.§c") 
-		element = {element}
-	end
-    assert(type(element) == "table", "§4your element table seems to to be incorrect.§c")
-    
+
+---SMOOTH HEAD - Mimics a vanilla player head, but smoother and with some extra life. Can also do smooth Torsos and Smooth Necks!
+---@param element ModelPart|table<ModelPart> The head element that you wish to effect. If you want a smooth neck or torso, instead of a single element, input a table of head elements(imagine it like {element1, element2, etc.}). This will apply the head rotations to each of these.
```

Comment:
`table<K, V>` is a type that takes in two parameters. If you meant for this to be an array of parts, use `ModelPart[]` instead.
```suggestion
---@param element ModelPart|ModelPart[] The head element that you wish to effect. If you want a smooth neck or torso, instead of a single element, input a table of head elements(imagine it like {element1, element2, etc.}). This will apply the head rotations to each of these.
```